### PR TITLE
Implementing optional parameter for all endpoints

### PR DESCRIPTION
**Pull request by sam-i-am012**
Addresses #29

### CODE REVIEW

**Comment by trel** [comment]
consider separating the .idea files removal from the optional parameter commits.

not a blocker.

File: src/main/java/org/example/Operations/CollectionOperations.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public Response create(String token, String lpath, OptionalInt createIntermediat
          formData.put("op", "create");
          formData.put("lpath", lpath);
          if (createIntermediates.isPresent()) {
-             formData.put("create-intermediates", String.valueOf(createIntermediates));
+             formData.put("create-intermediates", String.valueOf(createIntermediates.getAsInt()));
```
Comment (korydraughn) [inline src/main/java/org/example/Operations/CollectionOperations.java:36]: I noticed some lines use `.ifPresent` while others use an if-stmt.

Let's pick one style and stick with that unless there's a good reason to use a different style. _This is only a recommendation._

File: src/main/java/org/example/Properties/Ticket/TicketCreateParams.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+public class TicketCreateParams {
+    Optional<String> type = Optional.empty();
+    OptionalInt useCount = OptionalInt.empty();
+    OptionalInt writeDataObjectCount = OptionalInt.empty();
+    OptionalInt writeByteCount = OptionalInt.empty();
+    OptionalInt secondsUntilExpiration = OptionalInt.empty();
+    Optional<String> users = Optional.empty();
+    Optional<String> groups = Optional.empty();
+    Optional<String> hosts = Optional.empty();
```
Comment (korydraughn) [inline src/main/java/org/example/Properties/Ticket/TicketCreateParams.java:14]: Looks like you're missing the `private` access specifier?

Please double check other types like this.

File: src/main/java/org/example/Properties/Zone/ZoneAddParams.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package org.example.Properties.Zone;
+
+import java.util.Optional;
+
+public class ZoneAddParams {
+    Optional<String> connectionInfo = Optional.empty();
+    Optional<String> comment = Optional.empty();
```
Comment (korydraughn) [inline src/main/java/org/example/Properties/Zone/ZoneAddParams.java:7]: Missing `private`?

File: src/main/java/org/example/Operations/CollectionOperations.java (hunk 1 above)
Comment (sam-i-am012) [reply src/main/java/org/example/Operations/CollectionOperations.java:36]: When I went changed the way I implemented optional params I completely forgot that I was using `.ifPresent`. I've changed everything to use `.ifPresent` now.

File: src/main/java/org/example/Properties/Ticket/TicketCreateParams.java (hunk 2 above)
Comment (sam-i-am012) [reply src/main/java/org/example/Properties/Ticket/TicketCreateParams.java:14]: Oops I did forget it. It's been added to all places that was missing it.

File: src/main/java/org/example/Properties/Zone/ZoneAddParams.java (hunk 3 above)
Comment (sam-i-am012) [reply src/main/java/org/example/Properties/Zone/ZoneAddParams.java:7]: Added.

**Comment by sam-i-am012** [comment]
I've resolved all existent comments and am awaiting further review.

**Comment by korydraughn** [comment]
Is everything as you expect it?
Is there anything else left to do for this PR?

**Comment by korydraughn** [comment]
And do the tests pass?

**Comment by sam-i-am012** [comment]
Yep, all tests that are currently present pass. That should be it for this PR.

**Comment by korydraughn** [comment]
Squash to taste and pound it.

**Comment by korydraughn** [comment]
I see one commit is missing an issue number.

**Comment by sam-i-am012** [comment]
Done.

**Comment by korydraughn** [comment]
Excellent.

Merging.

**Pull request merged**